File: onion-architecture/java/com/example/domain/Course.java
package com.example.domain;

public class Course {
    private Long id;
    private String title;

    public Course(Long id, String title) {
        this.id = id;
        this.title = title;
    }

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    // equals/hashCode based on ID
    @Override
    public boolean equals(Object o) {
        return o instanceof Course && ((Course) o).id.equals(this.id);
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }
}

File: onion-architecture/kotlin/src/main/kotlin/com/example/domain/Course.kt
package com.example.domain

data class Course(
    val id: Long,
    val title: String
) {
    // Custom equals/hashCode based on ID only (like Java version)
    override fun equals(other: Any?): Boolean {
        return other is Course && other.id == this.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}